prov/rxm/src/rxm_init.c

```diff
@@ -66,33 +63,17 @@ int rxm_info_to_core(uint32_t version, const struct fi_info *hints,
 		if (hints->caps & (FI_MSG | FI_TAGGED))
 			core_info->caps |= FI_RMA;
 
-		/* No fi_info modes apart from those handled earlier in
-		 * this function can be passed along to the core provider */
-
-		if (hints->domain_attr) {
-			if (FI_VERSION_GE(version, FI_VERSION(1, 5))) {
-				/* Allow only those mr modes that can be
-				 * passed along to the core provider */
-				core_info->domain_attr->mr_mode |=
-					hints->domain_attr->mr_mode &
-					OFI_MR_BASIC_MAP;
-			} else if (ofi_rma_target_allowed(hints->caps)) {
-				core_info->domain_attr->mr_mode =
-					hints->domain_attr->mr_mode;
```

Comment:
We shouldn't remove this for the same reason in the previous comment. If app is RMA target it needs to support appropriate modes to handle a core provider requiring FI_MR_BASIC